db/migrate/20150831193037_move_observations_to_roles.rb

```diff
@@ -5,11 +5,21 @@ def change
 
     # We have to use raw sql because we've subclassed the observation class
     pgres = ProposalRole.connection.execute("SELECT proposal_id, user_id from observations")
+    seen_proles = {}
```

Comment:
thanks @cmc333333. I was not familiar with Ruby Set class. (My Perl background showing here, which doesn't have a built-in Set type.) 

Just read up on it. I like that the syntax is clearer. http://blog.x-aeon.com/2012/02/07/using-data-sets-in-ruby-performance-of-hash-vs-set/ has an interesting comparison.

Re: SQL suggestions, yes I see the advantages. The code was easier for me to write than the SQL, but I can rewrite if you think the timeouts are going to be an issue.